rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleParser.java

```diff
@@ -680,19 +698,31 @@ protected Literal parseQuotedLiteral() throws IOException, RDFParseException, RD
 			// Read datatype
 			Value datatype = parseValue();
 			if (datatype instanceof IRI) {
-				return createLiteral(label, null, (IRI) datatype, getLineNumber(), -1);
-			} else {
+				return createLiteral(label, null, (IRI)datatype, getLineNumber(), -1);
+			}
+			else if (datatype != null) {
```

Comment:
I don't know why the Eclipse code formatter keeps throwing these curve balls, but the important bit of the change is here and in lines 707-719.